030-extract-core-resources/proposal.md

```diff
@@ -0,0 +1,79 @@
+# Summary
+
+Concourse currently ships with the following "core" resource types:
+
+* [`bosh-io-release`](https://github.com/concourse/bosh-io-release-resource)
+* [`bosh-io-stemcell`](https://github.com/concourse/bosh-io-stemcell-resource)
+* [`cf`](https://github.com/concourse/cf-resource)
+* [`docker-image`](https://github.com/concourse/docker-image-resource)
+* [`git`](https://github.com/concourse/git-resource)
+* [`github-release`](https://github.com/concourse/github-release-resource)
+* [`hg`](https://github.com/concourse/hg-resource)
+* [`mock`](https://github.com/concourse/mock-resource)
+* [`pool`](https://github.com/concourse/pool-resource)
+* [`registry-image`](https://github.com/concourse/registry-image-resource)
+* [`s3`](https://github.com/concourse/s3-resource)
+* [`semver`](https://github.com/concourse/semver-resource)
+* [`time`](https://github.com/concourse/time-resource)
+* [`tracker`](https://github.com/concourse/tracker-resource)
+
+This proposal is to remove them all from the `.tgz` distribution, leaving only the `registry-image` resource, as it will be necessary for fetching other resource types.
```

Comment:
Have you considered offline environments? We are using Concourse in 3 air-gapped environments on AWS, which heavily rely on S3 since they can't reach the public docker registry. The change proposed here would require us to maintain a docker registry.

Reply:
@jvshahid do you use any resource types other than the core resource types? Wouldn't you therefore need a docker registry anyway to use any non-core resource types?

Additionally - if you can use S3 in your environment, can you use ECR (AWS's managed Docker registry)?

Reply:
We use the S3 resource. To be more specific, this is a US Federal user deploying to a separate partition of AWS. I am not sure if ECR is available in that region/partition.

Reply:
@jvshahid For what it's worth, the [AWS Region Table](https://aws.amazon.com/about-aws/global-infrastructure/regional-product-services/) does claim that ECR is available in US GovCloud (both West and East).

Reply:
Even if services are provider, we've have users that are not permitted to use these services. For instance, a user cannot host docker images because it requires security scanning and can slow their development of pipelines. The scanning of multiple pieces of software can take longer than one large one.

Reply:
I think we could make it easy to install a resource type archive into a local directory for the worker to load up, which is exactly what's happening today under the hood - it's just automated on start, and not documented for public use. Would that be sufficient for these use cases? Operators would just need to grab the archives (e.g. from our GitHub releases) and install them to their cluster.

Reply:
I believe that would work.

Reply:
We relied on this technique in the past to add custom resource types to BOSH deployed Concourse. This method broke when Concourse moved the resource types from `/var/vcap/packages/foo-resource-type` to `/var/vcap/packages/concourse/resource-types/foo`. The new directory structure makes it weird to add new custom types since we will have to modify the Concourse package on disk. 

I am mentioning this to make sure the extension points proposed above don't suffer from similar drawbacks.

Reply:
@jvshahid Thanks, will keep that in mind! I think if we can get the `containerd` track of work wrapped up before this, we might be able to make it as simple as placing OCI image tarballs in a directory. 🤞